TSQL_Parser/Tests/Statements/MergeStatementTests.cs

```diff
@@ -0,0 +1,352 @@
+﻿using NUnit.Framework;
+using System.Collections.Generic;
+using TSQL;
+using TSQL.Statements;
+
+namespace Tests.Statements
+{
+	[TestFixture(Category = "Statement Parsing")]
+	public class MergeStatementTests
+	{
+		[Test]
+		public void MergeStatement_StandardMerge_UsingValues()
+		{
+			List<TSQLStatement> statements = TSQLStatementReader.ParseStatements(
+				@"
+				MERGE INTO [t].[a] AS [Target]
+				USING (VALUES
+				  (1,N'v1',NULL)
+				 ,(2,N'v2',NULL)
+				) AS [Source] ([ID],[Val])
+				ON ([Target].[ID] = [Source].[ID])
+				WHEN MATCHED AND (
+					NULLIF([Source].[Val], [Target].[Val]) IS NOT NULL OR NULLIF([Target].[Val], [Source].[Val]) IS NOT NULL) THEN
+				 UPDATE SET
+				  [Target].[Val] = [Source].[Val],
+				WHEN NOT MATCHED BY TARGET THEN
+				 INSERT([ID],[Val])
+				 VALUES([Source].[ID],[Source].[Val])
+				WHEN NOT MATCHED BY SOURCE THEN
+				 DELETE
+				OUTPUT
+				  $ACTION AS [Action]
+				 ,DELETED.[ID]
+				 ,DELETED.[Val]
+				 ,INSERTED.[ID]
+				 ,INSERTED.[Val];",
+				includeWhitespace: true);
+			TSQLMergeStatement merge = statements[0] as TSQLMergeStatement;
+
+			Assert.IsNotNull(statements);
+			Assert.AreEqual(1, statements.Count);
+			Assert.AreEqual(TSQLStatementType.Merge, statements[0].Type);
+			Assert.AreEqual(194, merge.Tokens.Count);
+			Assert.AreEqual(TSQLKeywords.MERGE, merge.Tokens[0].AsKeyword.Keyword);
+			Assert.AreEqual(" ", merge.Tokens[1].AsWhitespace.Text);
+			Assert.AreEqual(TSQLKeywords.INTO, merge.Tokens[2].AsKeyword.Keyword);
+			Assert.AreEqual(" ", merge.Tokens[3].AsWhitespace.Text);
+			Assert.AreEqual("t", merge.Tokens[4].AsIdentifier.Name);
+			Assert.AreEqual(2, merge.Merge.Tokens.Count);
+			Assert.AreEqual(10, merge.Into.Tokens.Count);
+			Assert.AreEqual(33, merge.Using.Tokens.Count);

```

Comment:
There should be 34 tokens. The close parenthesis at position 410 is missing using the current parsing logic. The VALUES expression parsing inside the subquery parser is dropping the final paren. The problem is that the closing paren is not technically part of the VALUES expression, but it's the stop token.